keyboards/hineybush/hbcp/keymaps/default/keymap.c

```diff
@@ -0,0 +1,131 @@
+    KC_GRV,   KC_1,    KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,     KC_MINS, KC_EQL,  KC_BSPC, KC_NLCK, KC_PSLS, KC_PAST, KC_PAUS,
+    KC_TAB,   KC_Q,    KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,     KC_LBRC, KC_RBRC, KC_BSLS, KC_P7,   KC_P8,   KC_P9,   KC_PMNS,
+    KC_LGUI,  KC_A,    KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN,  KC_QUOT,          KC_ENT,  KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_LSFT,           KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,   KC_SLSH, KC_LSFT, KC_UP,   KC_P1,   KC_P2,   KC_P3,
+    KC_LCTL,  KC_LALT,                         KC_SPC,                                  MO(1),    KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT, KC_P0,   KC_PDOT, KC_PENT
+
+    ),
+
+  [1] = LAYOUT_wkl(
+    KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,           KC_PSCR, KC_SLCK,KC_HOME, KC_END,
+    KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,  RGB_HUD, RGB_HUI,KC_TRNS, KC_TRNS,
+    KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, RESET ,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,  RGB_SAD, RGB_SAI,KC_TRNS, KC_TRNS,
+    KC_CAPS,  KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS,          KC_TRNS,  RGB_VAD, RGB_VAI,KC_TRNS, KC_TRNS,
+    KC_TRNS,           KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_VOLU,  RGB_TOG, RGB_MOD,KC_TRNS,
+    KC_TRNS,  KC_TRNS,                         KC_MPLY,                                 KC_TRNS,  KC_TRNS, KC_MPRV, KC_VOLD,  KC_MNXT, KC_TRNS,KC_TRNS, KC_TRNS
+
+    ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QMKBEST:
+      if (record->event.pressed) {
+        // when keycode QMKBEST is pressed
+        SEND_STRING("QMK is the best thing ever!");
+      } else {
+        // when keycode QMKBEST is released
+      }
+      break;
+    case QMKURL:
+      if (record->event.pressed) {
+        // when keycode QMKURL is pressed
+        SEND_STRING("https://qmk.fm/" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
+
+void matrix_init_user(void) {
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+#ifdef RGBLIGHT_ENABLE
+// The first three LEDs are used as indicators for CAPS_LOCK, NUM_LOCK and SCROLL_LOCK.
+void keyboard_post_init_user(void) {
+    rgblight_set_effect_range(3, RGBLED_NUM-3);
+    led_set_user((1<<USB_LED_CAPS_LOCK)|(1<<USB_LED_NUM_LOCK)|(1<<USB_LED_SCROLL_LOCK));
+    wait_ms(300);
+    led_set_user(0);
+}
+
+
+void led_set_user(uint8_t usb_led) {
```

Comment:
move this up to the keyboard level and use `-__attribute__ ((weak))` then it can be overriden at the keymap level. Then it work correctly in configurator too.

Reply:
same thing for keyboard_post_init_user